Revise langauge and troubleshooting guide for working with resource mappings files (#751)
o3de/o3de.org: content/docs/user-guide/gems/reference/aws/aws-core/resource-mapping-files.md

toc: true
---

> A **resource mapping file** is a JSON file that provides a map of **lookup names** to a collection of **attributes** about AWS resources. O3DE runtimes can query the mapping file to easily find and utilize AWS resources. Usually different deployments, and consequently different mappings, are used for game development and for the released version of a game.

Reviewer: Update to use active voice, and switch to the more generic "project" instead of "game", per convention in these docs.
```suggestion
A **resource mapping file** is a JSON file that provides a map of **lookup names** to a collection of **attributes** about AWS resources. O3DE runtimes can query the mapping file to easily find and utilize AWS resources. Typically, a project under development will use a different deployment, and therefore a different mapping, than the released version of the project.
```

A mapping file contains one or more resource entries and a set of global attributes.

**Resource entry format**

Each entry defines a lookup name - a logical name for the resource - and a collection of attributes in the following format:

```json
"<lookup name>":  {
    "Type": string, (required)
    "Name/ID": string, (required)
    "Region": string, (optional)
    "AccountId": string (optional)
}
```

> ***Supported attributes***

Reviewer: ```suggestion
**Supported attributes**

```

Author: The intent here is for this to be nested inside the **resource entry format** section as its a subsection of that piece.

Reviewer: Using three asterisks in Markdown results in bold+italic formatting. Is that what you wanted to use to indicate nesting?

* Type - AWS resource type, e.g. AWS::Lambda::Function, AWS::S3::Bucket etc. Should map to a [CloudFormation resource type](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html).
* Name/ID - AWS resource name or id.

> * Region - Used to override the global region attribute if this resource is deployed to a different AWS region.

Reviewer 2: The field is mandatory when the global region is not set and it's used to construct the service URL.

Reviewer 3: global region attribute is mandatory as always, customer won't be able to skip setting global region attribute

Author: JSON schema defines global region as required, so it must be provided for the mapping to be valid.

}
```

This mapping file has a single mapping: **MyKey**, which is a Lambda function called "ExampleLambda". It is deployed in us-west-2 in the AWS account with account ID "123456789123".

## Generating a resource mappings file

The mapping files have a well-defined [JSON format](/docs/user-guide/gems/reference/aws/aws-core/resource-mapping-schema.md). They can be generated using custom tools or by hand. However, the easiest way to generate or edit these files is by using the visual [resource mapping tool](/docs/user-guide/gems/reference/aws/aws-core/resource-mapping-tool.md) provided with O3DE.

All mapping files should be saved into the project's `Config` directory.

## Using resource mappings

> The resource mapping file to use on load is controlled by the `ResourceMappingConfigFileName` setting in the `awscoreconfiguration.setreg` file; located in the project's `Registry` directory. See [Project Settings](./getting-started.md#project-settings) for more information about this file. The AWS Core Gem will read this mapping file and support access to the mappings.

Reviewer: Comma instead of semicolon, because the phrase after it is not a complete sentence.
```suggestion
The resource mapping file to use on load is controlled by the `ResourceMappingConfigFileName` setting in the `awscoreconfiguration.setreg` file, located in the project's `Registry` directory. See [Project Settings](./getting-started.md#project-settings) for more information about this file. The AWS Core Gem will read this mapping file and support access to the mappings.
```

Author: Fixed.
